### DISCUSSION TITLE
How do I get this left join to work?

### DISCUSSION THREAD

**Discussion by vinaymhetre** [supabase/supabase#21167] [Questions] [answered]
I want to run the equivalent query of this in javascript sdk. But I am unable to do so. What am I doing wrong?

Query:
select ec.id enrolled_course_id,* from courses c 
left outer join enrolled_courses ec
on c.id = ec.course_id
where c.name ilike '%react%'
and ec.id is null

Javascript:
const { data, error } = await supabase
    .from('courses')
    .select('*, enrolled_courses(id)')
    .ilike('name', `%react%`)
    .is('enrolled_courses.id', null)
    .limit(5);

The SQL works as expected but the javascript code still returns courses where there are matching enrolled_courses.

**Comment by GaryAustin1** [comment]
`.select('*, enrolled_courses!inner(id)')`

Adding the !inner clause will cause the main table to only show results with matching embedded table filters.

https://supabase.com/docs/reference/javascript/select?example=querying-referenced-table-with-inner-join

EDIT:  Just noticed you are doing outer join.

**Reply by vinaymhetre** [reply]
Yeah, doing inner join is straightforward when I need intersection. But I want to get records from courses that are not already present in enrolled_courses table.

**Reply by GaryAustin1** [reply, marked as answer]
You might look at this anti-join https://supabase.com/blog/postgrest-11-prerelease

Otherwise at some point a view or rpc call are better than continued playing around and figure out later if it can happen.

**Reply by vinaymhetre** [reply]
Oh! the anti-join seems to address my problem! Thanks a lot, really appreciate it. I will post an update if I find an edge case but for now marking your reply as an answer.
